components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/persistence/JDBCPersistenceManager.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.consent.management.common.persistence;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.bfsi.consent.management.common.config.ConsentManagementConfigParser;
+import org.wso2.bfsi.consent.management.common.exceptions.ConsentManagementRuntimeException;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+/**
+ * This class is used for handling consent management data persistence in the JDBC Store. During the server
+ * start-up, it checks whether the database is created, if not it creates one. It reads the data source properties
+ * from the bfsi-consent-management.xml. This is implemented as a singleton. An instance of this class can be obtained
+ * through JDBCPersistenceManager.getInstance() method.
+ */
+public class JDBCPersistenceManager {
+
+    private static volatile JDBCPersistenceManager instance;
+    private static volatile DataSource dataSource;
+    private static final Log log = LogFactory.getLog(JDBCPersistenceManager.class);
+
+    private JDBCPersistenceManager() {
+
+        initDataSource();
+    }
+
+    /**
+     * Get an instance of the JDBCPersistenceManager. It implements a double checked locking initialization.
+     *
+     * @return JDBCPersistenceManager instance
+     */
+    public static synchronized JDBCPersistenceManager getInstance() {
+
+        if (instance == null) {
+            synchronized (JDBCPersistenceManager.class) {
+                if (instance == null) {
+                    instance = new JDBCPersistenceManager();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Initialize the data source.
+     */
+    private void initDataSource() {
```

Comment:
Why isn't this a static synchronized method?

Reply:
initDataSource method is called only in the constroctor and the constructor is called through a static synchronized method. So we dont need to make this a static synchronized method